File: 프로그래머스/lv2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    stack = list()
    for element in s:
        if element == '(':
            stack.append(element)
        else:
            if not stack:
                return False
            if stack[-1] != '(':
                return False
            stack.pop()
    if not stack:
        return True
    return False

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.cpp
#include <string>
#include <stack>

using namespace std;

bool solution(string s) {
    stack<char> st;
    
    for (auto ch = s.begin(); ch != s.end(); ch++) {
        if (st.empty()) {
            if (*ch == ')') return false;
            st.push(*ch);
        } else {
            if (*ch == '(') st.push(*ch);
            else if (st.top() == '(') st.pop();
            else return false;
        }
    }
    
    return st.size() == 0;
}